rust/connlib/snownet/tests/lib.rs

```diff
@@ -1,37 +1,70 @@
 use boringtun::x25519::{PublicKey, StaticSecret};
+use firezone_relay::{AddressFamily, AllocationPort, ClientSocket, IpStack, PeerSocket};
+use rand::rngs::OsRng;
 use snownet::{Answer, ClientNode, Event, MutableIpPacket, ServerNode, Transmit};
 use std::{
     collections::HashSet,
     iter,
     net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4},
-    time::{Duration, Instant},
+    time::{Duration, Instant, SystemTime},
     vec,
 };
 use str0m::{net::Protocol, Candidate};
-use tracing::{info_span, Span};
+use tracing::{debug_span, info_span, Span};
 
 #[test]
-fn smoke() {
+fn smoke_direct() {
     let _ = tracing_subscriber::fmt()
         .with_test_writer()
         .with_env_filter("debug")
         .try_init();
 
     let (alice, bob) = alice_and_bob();
 
-    let mut alice = TestNode::new(info_span!("Alice"), alice);
-    alice.add_local_host_candidate("1.1.1.1:80");
+    let mut alice =
+        TestNode::new(info_span!("Alice"), alice, "1.1.1.1:80").with_local_host_candidate();
+    let mut bob = TestNode::new(info_span!("Bob"), bob, "1.1.1.2:80").with_local_host_candidate();
+    let firewall = Firewall::default();
+    let mut clock = Clock::new();
 
-    let mut bob = TestNode::new(info_span!("Bob"), bob);
-    bob.add_local_host_candidate("1.1.1.2:80");
+    handshake(&mut alice, &mut bob, &[], &clock);
 
-    handshake(&mut alice, &mut bob);
+    loop {
+        if alice.is_connected_to(&bob) && bob.is_connected_to(&alice) {
+            break;
+        }
+
+        progress(&mut alice, &mut bob, &mut [], &firewall, &mut clock);
```

Comment:
Without the `?` or `unwrap` I didn't notice at first that the clock will panic if the test runs past 1 virtual hour.

Reply:
Yep, that is the idea! Let's see how this pans out :)

Reply:
Should I panic further up?

Reply:
Nah it's fine. Even if it said `unwrap` I probably wouldn't immediately think that it was a timeout, even though it makes sense for a timeout to go there.